src/components/SingleToDo.vue

```diff
@@ -71,22 +67,54 @@
 
 <script setup lang="ts">
 import PrioSelector from '../components/PrioSelector.vue';
-import CheckTodo from './checkTodo.vue';
+import CheckTodo from './CheckTodo.vue';
 import SaveTodo from '../components/SaveTodo.vue';
 import { Todotype } from '../Types/toDo';
 
-const props = defineProps<{ todos: Todotype[]; checkedTodos: Todotype[] }>();
+let editingTodo: Todotype | null = null;
+
+const props = defineProps<{
+  todos: Todotype[];
+  singleEditTodo: boolean;
+  checkedTodos: Todotype[];
+}>();
 const emit = defineEmits<{
   (e: 'toggleEdit', toDo: Todotype, checkedTodo: Todotype): void;
   (e: 'setCheckedTodos', toDo: Todotype): void;
   (e: 'saveTodo', toDo: Todotype): void;
   (e: 'taskIndex', index: number): void;
   (e: 'notCheckedTodos', toDo: Todotype): void;
   (e: 'deleteCheckedTodos', index: number): void;
+  (e: 'editingFalse'): void;
 }>();
 
-function toggleEdit(checkedTodo: Todotype) {
-  checkedTodo.isEditing = true;
+function toggleEdit(toDo: Todotype) {
+  if (!props.singleEditTodo) {
+    toDo.isEditing = true;
+    emit('editingFalse');
+  }
+}
+
+function handleClickInside(toDo: Todotype) {
+  event.stopPropagation();
+  if (toDo.isEditing) {
+    return;
+  }
+
+  if (editingTodo && editingTodo !== toDo) {
+    editingTodo.isEditing = false;
+  }
+
+  if (props.checkedTodos && props.checkedTodos.includes(toDo)) {
```

Comment:
props.checkedTodos?.includes(toDo) change the condition to this, it will be the same. Here is the explanation: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining